src/librustc_mir/dataflow/move_paths/builder.rs

```diff
@@ -361,17 +361,14 @@ impl<'b, 'a, 'tcx> Gatherer<'b, 'a, 'tcx> {
     fn gather_terminator(&mut self, term: &Terminator<'tcx>) {
         match term.kind {
             TerminatorKind::Goto { target: _ }
+            | TerminatorKind::Return
             | TerminatorKind::Resume
             | TerminatorKind::Abort
             | TerminatorKind::GeneratorDrop
             | TerminatorKind::FalseEdges { .. }
             | TerminatorKind::FalseUnwind { .. }
             | TerminatorKind::Unreachable => {}
 
-            TerminatorKind::Return => {
-                self.gather_move(Place::return_place());
```

Comment:
So the reason this is ok, is because the MIR return terminator is actually a nop from a memory perspective. All the value returning should have already happened by assigning to the return place?

If this is correct, please leave it as a comment on the above `TerminatorKind::Return`.

Reply:
It's OK because there can't be any statements after a return to notice what we do here. I've added a comment.